### ISSUE TITLE
Audits does not seem to be Thread safe

### ISSUE THREAD

**Issue by ar45** [Atomidata/django-audit-log#31]
Hi,

I was looking for an audit log implementation and came across this project. However I do not understand why this would work in all cases. From what I understand, signals are managed and stored in memory and is the same for all threads. So if multiple requests come in at the same time you may end up with the signals from request 1 and signals from request 2 both being called for each request. 

Simply put. This is not thread safe, and disconnecting and reconnecting signals at runtime is a bad idea.

**Comment by vvangelovski** [comment]
Yes. And have you made any tests to reproduce your claim for this project in particular.

And if you don't "disconnect and reconnect signals at runtime" at what time do you do it?
